src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java

```diff
@@ -73,4 +90,156 @@
      * @return a {@link CompletableFuture} that completes when the request has been delivered to AxonServer
      */
     CompletableFuture<Void> mergeEventProcessor(String eventProcessorName, String tokenStoreIdentifier);
+
+    /**
+     * Request to create Axon Server user.
+     * Returns a {@link CompletableFuture} that completes when the request has been processed by AxonServer.
+     *
+     * @param request the request to create the user, containing the username, password and user roles
+     * @return a {@link CompletableFuture} that completes when the request has been processed to AxonServer
+     */
+    CompletableFuture<Void> createOrUpdateUser(CreateOrUpdateUserRequest request);
+
+    /**
+     * Request to list all users in Axon Server.
+     *
+     * @return a {@link CompletableFuture} that contains list of all users in Axon Server.
+     */
+    CompletableFuture<List<UserOverview>> getAllUsers();
+
+    /**
+     * Request to delete user in Axon Server.
+     *
+     * @param username is the username of the user to delete
+     * @return a {@link CompletableFuture} that completes when user has been deleted
+     */
+    CompletableFuture<Void> deleteUser(String username);
+
+    /**
+     * Request to create Axon Server application.
+     * Returns a {@link CompletableFuture} that completes when the request has been processed by AxonServer.
+     *
+     * @param request the request to create the application, containing the application name, the application roles...
+     * @return a {@link CompletableFuture} that completes when the request has been processed to AxonServer
+     */
+    CompletableFuture<Token> createOrUpdateApplication(ApplicationRequest request);
+
+    /**
+     * Request to retrieve application overview in Axon Server.
+     *
+     * @param applicationName is the name of the application to retrieve
+     * @return a {@link CompletableFuture} containing application data such as name, roles, etc.
+     */
+    CompletableFuture<ApplicationOverview> getApplication(String applicationName);
+
+    /**
+     * Request to refresh token of application in Axon Server.
+     *
+     * @param applicationName is the name of the application to refresh token
+     * @return a {@link CompletableFuture} containing new token
+     */
+    CompletableFuture<Token> refreshToken(String applicationName);
+
+    /**
+     * Request to delete application in Axon Server.
+     *
+     * @param applicationName is the name of the application to delete
+     * @return a {@link CompletableFuture} that completes when application has been deleted
+     */
+    CompletableFuture<Void> deleteApplication(String applicationName);
```

Comment:
I left it on _delete_ to match previous REST API in Axon Server, also UI button has hint "delete". Not sure of the difference, but it might be that delete is right term, as something that is removed can be re-added, but deleted can't, as its case with application -> token is permanently destroyed